boards/common/nrf52xxxdk: Fixed periph conf

maribu opened the pull request:
### Contribution description
- Used proper `GPIO_PIN()` macro to declare pins, instead of hard coded values
- Allow providing a board specific SPI config by changing the common config to only provide an SPI config if none has been supplied yet
- Added board specific SPI config for the nRF52-DK, as it differs from the common config
- Added board specific SPI config for the nRF52840-DK, as it differs from the common config

### Testing procedure

E.g. run `examples/gnrc_networking` with the following patch applied and the W5100 ethernet shield.

```patch
diff --git a/examples/gnrc_networking/Makefile b/examples/gnrc_networking/Makefile
index 78a9eb7f56..29d3ae3da3 100644
--- a/examples/gnrc_networking/Makefile
+++ b/examples/gnrc_networking/Makefile
@@ -19,7 +19,6 @@ BOARD_INSUFFICIENT_MEMORY := arduino-duemilanove arduino-mega2560 arduino-nano \
 
 # Include packages that pull up and auto-init the link layer.
 # NOTE: 6LoWPAN will be included if IEEE802.15.4 devices are present
-USEMODULE += gnrc_netdev_default
 USEMODULE += auto_init_gnrc_netif
 # Activate ICMPv6 error messages
 USEMODULE += gnrc_icmpv6_error
@@ -40,6 +39,7 @@ USEMODULE += ps
 USEMODULE += netstats_l2
 USEMODULE += netstats_ipv6
 USEMODULE += netstats_rpl
+USEMODULE += w5100
 
 # Comment this out to disable code in RIOT that does safety checking
 # which is not needed in a production environment but helps in the
@@ -74,3 +74,7 @@ else
     CFLAGS += -DIEEE802154_DEFAULT_CHANNEL=$(DEFAULT_CHANNEL)
   endif
 endif
+
+CFLAGS += '-DW5100_PARAM_CS=GPIO_PIN(1,12)'
+CFLAGS += '-DW5100_PARAM_EVT=GPIO_PIN(1,3)'
+CFLAGS += '-DDEBUG_ASSERT_VERBOSE'
```

### Issues/PRs references

None

------

Update: Adapted description to reflect changes on this PR based on the received review

maribu commented:
I was so free to fix the missing doc of `SPI_PORT` and squash, as noone started to review yet

maribu commented:
@aabadie: Do you have time to review this PR? It is rather trivial.

aabadie commented:
I'll give this a try tomorrow

maribu commented:
@aabadie: Already had a chance to test this?

maribu commented:
@aabadie: Thanks for the review. I adapted the PR to split off the changes of using GPIO_PIN() to define the pins into a separate commit, as requested. I also got rid of the `NRF52XXXDK_SPI_PORT` macro. Instead, I changed the common header to only provide the SPI config if no board specific config has been provided yet. Finally, I added the board specific config of both the nRF52840-DK and the nRF52-DK (as separate commit each).

maribu commented:
Btw: I didn't move the SPI config out of `nrf52xxxdk` as it is also used used by the `ruuvitag` and the `acd52842` board, and I don't know what the correct SPI config for those two would be.

aabadie commented:
Murdock is reporting issues with nrf52840dk, I think there's an `#include "periph_cpu.h"` missing in the board `periph_conf.h` or something. Feel free to squash directly.

aabadie commented on boards/common/nrf52xxxdk/include/periph_conf_common.h:
```diff
@@ -34,16 +34,18 @@ extern "C" {
  * @name    SPI configuration
  * @{
  */
+#ifndef SPI_NUMOF /* <-- Allow overwriting SPI_NUMOF from periph_conf */
```
What about moving this spi config in its own default include file in common nrf52 directory (like cfg_rtt_default.h) ?
Only ruuvitag and the other board would include it in their `periph_conf.h` and you can remove it from this file. And then no to define the structure before including periph_conf_common in nrf52dk/nrf52840dk.

Do this makes sense for you @maribu ?

maribu commented:
@aabadie: I split off the SPI config to `cfg_spi_default.h` and put that into `boards/common/nrf52` instead of `boards/common/nrf52xxxdk`, as

1. This default SPI config might useful to any nRF52 based boards, not just DKs
2. None of the actual DK boards uses it

aabadie commented:
> None of the actual DK boards uses it

but MDKs do :), see nrf52832-mdk and nrf52840-mdk.

maribu commented:
May I rebase to current master, so that I can update the `nrf52832-mdk` as well? I would still keep the default SPI config in `boards/common/nrf52`, agreed?

aabadie commented:
> May I rebase to current master, so that I can update the nrf52832-mdk as well?

Since it's in master and uses the same default spi config, yes, definitely. That will remove another example of code duplication.

maribu commented:
Hmm, the `ruuvitag` and the `acd52830` don't use the common periph anyway. I just assumed that would be the reason for the to depend on `boards/common/nrf52xxxdk`. Maybe that dependency is an artifact and could be removed?

aabadie commented:
> I just assumed that would be the reason for the to depend on boards/common/nrf52xxxdk. Maybe that dependency is an artifact and could be removed?

I just had a look at the code and for acd52xx, I think it's ok. For ruuvitag, it uses the common/nrf52xxxdk common module because of board.c.
I would say that there's no need to change them.

In the current state, I would ACK this PR

maribu commented:
@aabadie: It seems [Github sometimes fails to handle rebases properly](https://help.github.com/en/articles/why-are-my-commits-in-the-wrong-order), this is the correct order of commits (from most recent to the oldest):

```
commit c6bd64e3a4ef86cc5795c14d0087ad8ec98b1888
Author: Marian Buschsieweke <marian.buschsieweke@ovgu.de>
Date:   Wed Apr 10 16:43:27 2019 +0200

    boards/nrf52840dk: Arduino-compatible SPI config
    
    Added a SPI config that allows using Arduino-Shields that use SPI via the
    Arduino-ICSP header

commit ccd7679a3b66e614693f9a8485dbe37d50ad55a6
Author: Marian Buschsieweke <marian.buschsieweke@ovgu.de>
Date:   Wed Apr 10 16:43:16 2019 +0200

    boards/nrf52dk: Arduino-compatible SPI config
    
    Added a SPI config that allows using Arduino-Shields that use SPI via the
    Arduino-ICSP header

commit 2d022795f4c422b2b38eb249c5f467e47c351b55
Author: Marian Buschsieweke <marian.buschsieweke@ovgu.de>
Date:   Thu Apr 11 09:35:40 2019 +0200

    boards/nrf52832-mdk: Use default SPI config

commit 1280bf3d386a6a427b680872136a0743cf688abb
Author: Marian Buschsieweke <marian.buschsieweke@ovgu.de>
Date:   Thu Apr 11 09:35:16 2019 +0200

    boards/nrf52840-mdk: Use default SPI config

commit be061d76a196d93a9b34932a0969373e50299d40
Author: Marian Buschsieweke <marian.buschsieweke@ovgu.de>
Date:   Wed Apr 10 16:33:29 2019 +0200

    boards/common/nrf52xxxdk: Split off SPI config
    
    Moved SPI config from boards/common/nrf52xxxdk/include/periph_config.h to
    boards/common/nrf52/include/cfg_spi_default.h. This allows all nRF52 based
    boards to use this SPI config if applicable and also allows nRF52*-DK boards
    to use custom SPI configs, if needed.

commit 4d7426e13ca689eb26f5cbc1a20dec8e9d5d2ea9
Author: Marian Buschsieweke <marian.buschsieweke@ovgu.de>
Date:   Wed Apr 10 16:24:42 2019 +0200

    boards/common/nrf52xxxdk: Use GPIO_PIN in SPI conf
```

If I read the help article above correctly, this should not be a problem for merging.

maribu commented:
@aabadie: I rebased to fix the merge conflict with https://github.com/RIOT-OS/RIOT/pull/11513.

aabadie commented:
@maribu, thanks for reviving this PR, I forgot about it.

Codewise I'm now fine with the changes. I'll give it a try.
